src/main/java/org/jabref/gui/BasePanel.java

```diff
@@ -747,16 +749,17 @@ public void setupMainPanel() {
         splitPane.getItems().add(pane);
 
         // Set up name autocompleter for search:
-        instantiateSearchAutoCompleter();
-        this.getDatabase().registerListener(new SearchAutoCompleteListener());
-
-        setupAutoCompletion();
-
+        executorService.execute(() -> {
```

Comment:
I would leave the `registerListener` call on the current thread (just because I'm not sure how listeners interact with multithreading in detail - and this is no expensive operation)

Reply:
Fixed